src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -11,7 +11,9 @@
 SyncHttpClient = com.puppetlabs.http.client.Sync
 
 class Puppet::Server::HttpClient
+
   attr_reader :client
+  @@client = nil
```

Comment:
I think we can get away without having to use this `@@` variable here; see subsequent comments.  I've had pretty bad experiences with those variables in the past.